src/v/datalake/coordinator/catalog_factory.cc

```diff
@@ -76,11 +135,17 @@ std::unique_ptr<iceberg::catalog> create_catalog(
           "Creating rest Iceberg catalog connected to ",
           cfg.iceberg_rest_catalog_endpoint().value());
 
+        auto addr = endpoint_to_address(
+          cfg.iceberg_rest_catalog_endpoint().value());
         std::unique_ptr<http::abstract_client> http_client
           = static_cast<std::unique_ptr<http::abstract_client>>(
             std::make_unique<http::client>(net::base_transport::configuration{
               .server_addr = endpoint_to_address(
-                cfg.iceberg_rest_catalog_endpoint().value())}));
+                cfg.iceberg_rest_catalog_endpoint().value()),
+              .credentials = co_await build_tls_credentials(
+                std::move(metric_detail), std::nullopt, datalake_log),
+              .tls_sni_hostname = addr.host(),
```

Comment:
can this be passed if we are not using HTTPS ?

Reply:
Maybe we can return more information from `endpoint_to_address` to determine if we should use TLS